Framework/TestHelpers/src/ComponentCreationHelper.cpp

```diff
@@ -603,12 +603,99 @@ Instrument_sptr createTestInstrumentRectangular(int num_banks, int pixels,
  * Pixels are 4 mm wide.
  *
  * @param num_banks: number of rectangular banks to create
- * @param pixels :: number of pixels in each direction.
- * @param pixelSpacing :: padding between pixels
+ * @param pixels : number of pixels in each direction.
+ * @param nameless : components are unnamed if true.
+ * @param pixelSpacing : padding between pixels
  */
 Instrument_sptr createTestInstrumentRectangular2(int num_banks, int pixels,
+                                                 bool nameless,
                                                  double pixelSpacing) {
-  auto testInst = boost::make_shared<Instrument>("basic_rect");
+
+  auto instrName = nameless ? "" : "basic_rect";
```

Comment:
Check for emptiness with `empty()` method.

Reply:
here i am assigning to an empty string